### PR TITLE
Fix pyi annotation for ProcessGroupNCCL.Options

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130957

Probably all the other options need updating too, but this is the one I
needed.  The accurate annotation was determined by reading
torch/csrc/distributed/c10d/init.cpp

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Review by wconstab — APPROVED** [review]

**Review by fduwjj — APPROVED** [review]

**Comment by fduwjj** [comment]
Yeah, We need to make this pyi more accurate and update-to-date..

File: torch/_C/_distributed_c10d.pyi [hunk 1]
```diff
@@ -544,19 +544,23 @@ class _ProcessGroupWrapper(Backend):
...
-        def _timeout(self, val: timedelta) -> None: ...
+    class NCCLConfig:
+        blocking: int
+        cga_cluster_size: int
+        min_ctas: int
+        max_ctas: int
+
+        def __init__(self) -> None: ...
         @property
-        def _is_high_priority_stream(self) -> bool: ...
```
Comment (shuqiangzhang) [inline torch/_C/_distributed_c10d.pyi:556]: Did you rebase before the change?

Comment (ezyang) [reply torch/_C/_distributed_c10d.pyi:556]: No, looks like I do need to rebase

File: torch/_C/_distributed_c10d.pyi [hunk 2]
```diff
@@ -544,19 +544,23 @@ class _ProcessGroupWrapper(Backend):
...
-        @_is_high_priority_stream.setter
-        def _is_high_priority_stream(self, val: bool) -> None: ...
+        def net_name(self) -> str: ...
 
+    class Options:
```
Comment (wanchaol) [inline torch/_C/_distributed_c10d.pyi:557]: did we delete the `_timeout` intentionally from NCCLOptions?

Comment (ezyang) [reply torch/_C/_distributed_c10d.pyi:557]: I checked the bindings and did not see timeout

**Comment by ezyang** [comment]
When I rebased, it looked like someone @shuqiangzhang had modified these annotations as part of adding the split_group API. However, I double checked the annotations and I like mine better (as they are more accurate, and they don't use the @property decorator unnecessarily).

**Pull request closed without merging**